### ISSUE TITLE
BSP2 will not load.  "Wait background load..."

### ISSUE THREAD

**Issue by mgabriell01** [aaFn/Bookmark-search-plus-2#295]
Please help.  For a few weeks or so, when I try to open BSP2, all I get is "Wait background load...", I tried restarting Firefox several times, and updated all installed extensions, but nothing helps.  Thanks in advance!

**Comment by mgabriell01** [comment]
Other extensions are working as expected.

**Comment by aaFn** [comment]
Hello @mgabriell01 , can you use that procedure here https://github.com/aaFn/Bookmark-search-plus-2/wiki/Getting-and-providing-logs-for-debugging-BSP2,
with executing the part "If this is a problem when loading BSP2 .." in step 5 ?

Thank you, aaFn.

**Comment by mgabriell01** [comment]
Hello, aaFn.  Below is the whole content of the browser console as you requested.  I look forward to your reply.  Thanks!
-------------------

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://airlytics-web.airlock.twcmobile.weather.com/eventproxy/track. (Reason: CORS request did not succeed). Status code: (null).
ExtensionPreferencesManager API ftpProtocolEnabled created but addSetting was not called.
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://airlytics-web.airlock.twcmobile.weather.com/eventproxy/track. (Reason: CORS request did not succeed). Status code: (null). 4
Image corrupt or truncated.
Image corrupt or truncated. URI in this note truncated due to length. data:AAABAAEAEBAAAAEAIAB4BAAAFgAAADgAAAAQAAAAIAAAAAEAIAAAAAAAAAAAABMLAAATCwAAAAAAAAAAAAAAAP8AAP8AAP8
Image corrupt or truncated.
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://airlytics-web.airlock.twcmobile.weather.com/eventproxy/track. (Reason: CORS request did not succeed). Status code: (null).
Load saved state..

background.js:143:10
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 9) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 8) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 7) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 6) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 5) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 4) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 3) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 2) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 1) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). libstore.js:471:16
uncaught exception: null
IndexedDB UnknownErr: ActorsParent.cpp:556
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.sys.mjs:841
libstore error on loading from local storage 3 : Error: An unexpected error occurred (retries left: 0) libstore.js:462:13
fileName:   undefined libstore.js:466:14
lineNumber: 0 libstore.js:467:14
Error on loading from local storage : undefined background.js:3573:10
uncaught exception: null
ExtensionPreferencesManager API ftpProtocolEnabled created but addSetting was not called.

-------------------



    On Friday, February 9, 2024 at 02:45:57 AM EST, aaFn ***@***.***> wrote:  
 
 


Hello @mgabriell01 , can you use that procedure here https://github.com/aaFn/Bookmark-search-plus-2/wiki/Getting-and-providing-logs-for-debugging-BSP2,
with executing the part "If this is a problem when loading BSP2 .." in step 5 ?

Thank you, aaFn.

**Comment by aaFn** [comment]
Hello @mgabriell01 , thank you.
Bad news : it looks like your Firefox local storage where BSP2 settings are stored is corrupt .. As seen so far, this does not happen often, but this happens, and the cause lies with Firefox.

The way to clean things up shoud be to reset the local storage for BSP2. This cannot be done by the add-on. It can only be done manually.
The only procedure I know of shoud be:
- Go to the add-ons page
- Uninstall BSP2 (this will remove the local storage for BSP2)
- Reinstall BSP2

Things shoud start working again, but you will have to re-configure BSP2 options, and BSP2 will refetch all favicons if that option is enabled.

Let me know how that goes.

**Comment by aaFn** [comment]
Hello, meanwhile, I found a way to internally clear the local storage, I do not know if this is as efficient as the method above, but I added it in 2.0.124.

Now, after 10 unsuccessful attempts to load the configuration from local storage, BSP2 will try to clear its local storage, to be able to restart.
This implies resetting all options to defaults, which is also what happens in the above procedure. And since the local storage is unreadable, these settings are lost forever anyway.
